core/src/impl/Kokkos_Combined_Reducer.hpp

```diff
@@ -183,10 +183,9 @@ KOKKOS_INLINE_FUNCTION auto _get_value_from_combined_reducer_ctor_arg(
 
 template <class T>
 KOKKOS_INLINE_FUNCTION auto _get_value_from_combined_reducer_ctor_arg(
-    T&& arg) noexcept ->
-    typename std::enable_if_t<is_view<std::decay_t<T>>::value,
-                              std::decay_t<T>>::value_type {
-  return arg();
+    T&&) noexcept -> typename std::enable_if_t<is_view<std::decay_t<T>>::value,
```

Comment:
```suggestion
    T&&) noexcept -> std::enable_if_t<is_view<std::decay_t<T>>::value,
```

Reply:
No, `typename` is required here. We use the dependent type `value_type` for whatever `std::enable_if_t` gives us.

Reply:
Sry